File: frontend/src/services/PaymentService.js
import axios from 'axios';

const PAYMENT_API_BASE_URL = "http://localhost:8080/payments";

class PaymentService {
    getAllPayments() {
        return axios.get(PAYMENT_API_BASE_URL);
    }

    getPaymentById(id) {
        return axios.get(`${PAYMENT_API_BASE_URL}/${id}`);
    }

    createPayment(payment) {
        return axios.post(PAYMENT_API_BASE_URL, payment);
    }

    updatePayment(id, payment) {
        return axios.put(`${PAYMENT_API_BASE_URL}/${id}`, payment);
    }

    deletePayment(id) {
        return axios.delete(`${PAYMENT_API_BASE_URL}/${id}`);
    }
}

export default new PaymentService();

File: src/main/java/Airlines/AirlineTicket/Payment/Models/PaymentService.java
package Airlines.AirlineTicket.Payment.Models;

import Airlines.AirlineTicket.Payment.Payment;

import java.util.List;

public interface PaymentService {

    Payment savePayment(Payment payment);

    List<Payment> getAllPayments();

    Payment getPaymentById(Long id);

    Payment updatePayment(Long id, Payment paymentDetails);

    void deletePayment(Long id);
}
